### PR TITLE
Packaging improvements

### PR DESCRIPTION
**Pull request by treydock**
Make file structure for deb package match RPM
Make files installed by deb and RPM packages pull from same file

### CODE REVIEW

File: lib/tasks/build_utils.rb [hunk 1]
```diff
@@ -212,6 +212,10 @@ def task_file(filename)
     "#{cwd}/files/#{filename}"
   end
 
+  def package_file(filename)
+    File.join(proj_root, 'packaging/files', filename)
+  end
```
Comment (johrstrom) [inline lib/tasks/build_utils.rb:217]: can we get rid of `task_file`?

Comment (treydock) [reply lib/tasks/build_utils.rb]: Yep, I removed

File: lib/tasks/packaging.rb [hunk 2]
```diff
@@ -37,17 +37,21 @@ def deb_build_cmd(packaging_dir, work_dir, output_dir, dist, version, extra_args
   desc "Tar and zip OnDemand into packaging dir"
   task :tar, [:dist] do |task, args|
     dist = args[:dist] || 'el8'
+    cmd = ['git', 'ls-files', '|', tar, '-c']
     if dist =~ /^el/
       version = ood_package_version
       tar_file = "packaging/rpm/v#{version}.tar.gz"
     else
       dir = File.join(Dir.pwd, 'build').tap { |p| sh "mkdir -p #{p}" }
       version = ood_package_version.gsub('-', '.')
```
Comment (johrstrom) [inline lib/tasks/packaging.rb:46]: WRT refactoring versions - it's kinda shown here in `version = ood_package_version.gsub('-', '.')` I just kinda don't think this knowledge belongs here.

Maybe we need some sort of raw `version` with `rpm_version` and `deb_version` that are derived from it?  In any case - it's about pushing this knowledge about what's valid for what packaging system etc somewhere more appropriate.

**Review by johrstrom — CHANGES_REQUESTED** [review]
Can we get test cases for the new files in `packaging/files`?

**Comment by treydock** [comment]
> Can we get test cases for the new files in `packaging/files`?

For some, they are tested by E2E like the Apache sudo.  For others I will have to add tests to E2E that validate the files exist with some amount of content checking.

File: lib/tasks/packaging.rb (hunk 2 above)
Comment (treydock) [reply lib/tasks/packaging.rb]: Added some better logic to handle versions and also same idea for nightly versions too.

**Comment by treydock** [comment]
Added some E2E tests for the various files deployed.  It lead me to discover some bugs in how I initially did `ExecReload` for Apache since the commands vary wildly between each OS, especially between EL and Ubuntu.  This will all become much easier to manage with #1547 using Rake task and Ruby templates to deploy these files.

**Comment by treydock** [comment]
Got E2E tests working with file tests.

**Review by johrstrom — APPROVED** [review]
looks good to me.

**Pull request merged**